apollo-ios-codegen/Sources/ApolloCodegenLib/Templates/SwiftPackageManagerModuleTemplate.swift

```diff
@@ -10,6 +10,8 @@ struct SwiftPackageManagerModuleTemplate: TemplateRenderer {
   let target: TemplateTarget = .moduleFile
 
   let config: ApolloCodegen.ConfigurationContext
+  
+  let version: ApolloCodegenConfiguration.SchemaTypesFileOutput.ApolloPackageVersion
```

Comment:
Can't this template get the version from the `config` internally? By adding this parameter, you're adding it to the synthesized initializer. Which means the call site has to access and pass in the `version`.

If you want this to be a storied property, I'd write a custom initializer that doesn't take in the `version`, just gets it from the `config` in the `init`.

Or just get the `version` from the `config` inside of the rendering function and kill the stored property altogether.

Reply:
We can do it that way yea, did it this way currently because the file generator is already handling the `switch` on the module type and getting the associated value, can move this logic into the template if you feel like thats the more appropriate place to do it.